### PR TITLE
Bump observability version for OpenSearch 1.2 release

### LINKED ISSUE #118: Release version 1.2
Coming from [opensearch-build#567](https://github.com/opensearch-project/opensearch-build/issues/567), release version 1.2. Please follow the following checklist.

### Preparation

- [x] Assign this issue to a release owner.
- [x] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
- [x] Create, update, triage and label all features and issues targeted for this release with `v1.2.0`.

### CI/CD

- [x] Increment version on main to `1.2.0.0`.
- [x] Ensure working and passing CI.
- [x] Re(add) this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/1.2.0).

### Pre-Release

- [x] Branch and build from a `1.2` branch.
- [x] Update your branch in the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/1.2.0).
- [x] Feature complete, pencils down.
- [x] Fix bugs that target this release.

### Release

- [x] Complete [documentation](https://github.com/opensearch-project/documentation-website).
- [x] Gather, review and publish release notes.
- [x] Verify all issued labeled for this release are closed or labelled for the next release.

### Post Release

- [x] Create [a release tag](https://github.com/opensearch-project/.github/blob/main/RELEASING.md#tagging).
- [ ] Suggest improvements to [this template](https://github.com/opensearch-project/opensearch-plugins/templates/release.md).
- [ ] Conduct a postmortem, and publish its results.

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
- bump version of frontend and backend plugins to 1.2.0.0
- rename observability frontend plugin artifact to `dashboards-observability`
- remove release workflow since we are using manifest files to build for release

### Issues Resolved
#118

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/draft-release-notes-config.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Setting the formatting and sorting for the release notes body
+name-template: Version $RESOLVED_VERSION
+change-template: "* $TITLE ([#$NUMBER](https://github.com/opensearch-project/sql/pull/$NUMBER))"
+sort-by: merged_at
+sort-direction: ascending
+replacers:
+  - search: '##'
+    replace: '###'
+
+# Organizing the tagged PRs into unified categories
```
Comment (eugenesk24) [inline .github/draft-release-notes-config.yml:15]: Should we be tagging our PRs?

**Review by eugenesk24 — APPROVED** [review]
lgtm :)

File: .github/draft-release-notes-config.yml (hunk 1 above)
Comment (joshuali925) [reply .github/draft-release-notes-config.yml:15]: that would be ideal

File: .github/draft-release-notes-config.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# The overall template of the release notes
+template: |
+  Compatible with OpenSearch and OpenSearch Dashboards Version $RESOLVED_VERSION
+  $CHANGES
+
+# Setting the formatting and sorting for the release notes body
+name-template: Version $RESOLVED_VERSION
+change-template: "* $TITLE ([#$NUMBER](https://github.com/opensearch-project/sql/pull/$NUMBER))"
```
Comment (ps48) [inline .github/draft-release-notes-config.yml:8]: Question: should this be sql link?

**Review by ps48 — APPROVED** [review]

File: .github/draft-release-notes-config.yml (hunk 2 above)
Comment (joshuali925) [reply .github/draft-release-notes-config.yml:8]: good point.. forgot to change it

File: dashboards-observability/common/constants/shared.ts [hunk 3]
```diff
@@ -27,7 +27,7 @@ export const PPL_ENDPOINT = '/_plugins/_ppl';
 export const SQL_ENDPOINT = '/_plugins/_sql';
 export const DSL_ENDPOINT = '/_plugins/_dsl';
 
-export const observabilityID = 'observability';
+export const observabilityID = 'observability-dashboards';
```
Comment (eugenesk24) [inline dashboards-observability/common/constants/shared.ts:30]: Why is this one observability-dashboards instead of dashboard-observability??

**Review by ps48 — APPROVED** [review]

File: dashboards-observability/common/constants/shared.ts (hunk 3 above)
Comment (joshuali925) [reply dashboards-observability/common/constants/shared.ts:30]: just to be consistent with other plugins. I think part of the reason is that it shows up in URL, which means it has to be a dashboards frontend plugin, but I don't think we have a standard on this

**Pull request merged**